### PR TITLE
server and network_sensor: frame transmission without serialization

### PR DESCRIPTION
**Pull request by rbudai98**
Frame serialization in case of quarter MP requires 2 miliseconds and in case of MP requires 8 miliseconds in plus

app/server: added 1 byte in front of every libwebsocket packet to determine if the message is serialized or not

network_sensor: after libwebsocket obtains the message the first byte is checked and based in it's value the following part is deserialized or not

Frame transmission without serialization finalized. (Needs testing on MP datasizes)

### CODE REVIEW

File: examples/first-frame-network/main.cpp [hunk 1]
```diff
@@ -104,14 +105,29 @@ int main(int argc, char *argv[]) {
...
+
+    while (true)
+    {
+        current_ticks = clock();
+        status = camera->requestFrame(&frame);
+
+        delta_ticks = clock() - current_ticks; //the time, in ms, that took to render the scene
+        if(delta_ticks > 0)
+            fps = CLOCKS_PER_SEC / delta_ticks;
+        std::cout << fps << std::endl;
```
Comment (dNechita) [inline examples/first-frame-network/main.cpp:120]: This is a good way to measure the duration of requestFrame() but should not be part of the first-frame-network example.

File: sdk/src/connections/network/network.cpp [hunk 2]
```diff
@@ -425,18 +428,40 @@ int Network::callback_function(struct lws *wsi,
...
+
+                /*Notify the host SDK that data is received from server*/
+                Cond_Var[connectionId].notify_all();
+            } else {
+                // process message of only frame data
+                uint8_t *m_frame = NULL;
+                int m_frameLength;
+                m_frameLength = static_cast<int>(len) - 1;
+
+                if (m_frame != NULL) {
```
Comment (dNechita) [inline sdk/src/connections/network/network.cpp:449]: you always set m_frame to NULL (just 4 lines above). This if() statement doesn't really make sense.

File: examples/first-frame-network/main.cpp (hunk 1 above)
Comment (rbudai98) [reply examples/first-frame-network/main.cpp]: Removed

File: sdk/src/connections/network/network.cpp (hunk 2 above)
Comment (rbudai98) [reply sdk/src/connections/network/network.cpp]: removed (not needed anymore)

File: sdk/src/connections/network/network.h [hunk 3]
```diff
@@ -73,6 +73,8 @@ class Network {
 
     static payload::ClientRequest send_buff[MAX_CAMERA_NUM];
     static payload::ServerResponse recv_buff[MAX_CAMERA_NUM];
+    uint8_t *m_frame;
```
Comment (SeptimiuVana) [inline sdk/src/connections/network/network.h:76]: is this variable used?

Comment (rbudai98) [reply sdk/src/connections/network/network.h:76]: removed

**Review by dNechita — APPROVED** [review]

**Pull request merged**